backend/coreapp/views.py

```diff
@@ -215,92 +217,42 @@ def fork(request, slug):
-                "error": "GitHub sign-in not configured"
-            }, status=status.HTTP_501_NOT_IMPLEMENTED)
-
-        required_params = ["code"]
-
-        for param in required_params:
-            if param not in request.data:
-                return Response({"error": f"Missing parameter: {param}"}, status=status.HTTP_400_BAD_REQUEST)
+    def get(self, request: Request):
+        return Response({
+            "user": serialize_profile(request, request.profile),
+        })
 
-        logging.debug("Attempting GitHub oauth login")
-
-        response = requests.post(
-            "https://github.com/login/oauth/access_token",
-            json={
-                "client_id": settings.GITHUB_CLIENT_ID,
-                "client_secret": settings.GITHUB_CLIENT_SECRET,
-                "code": request.data["code"],
-            },
-            headers={ 'Accept': 'application/json' },
-        ).json()
-
-        error = response.get("error")
-        if error == None:
-            access_token = response["access_token"]
-
-            profile.github_access_token = access_token
-
-            try:
-                assert profile.load_fields_from_github(always=True)
-                logging.debug("Connected existing profile to new GitHub user")
-            except GitHubUserChangeException:
-                # The token was for a different user than the one associated with the current profile,
-                # so make a new profile for this one.
-                profile = Profile()
-                profile.github_access_token = access_token
-                assert profile.load_fields_from_github(always=True)
-                logging.debug("Connected new profile to new GitHub user")
-            except GitHubUserHasExistingProfileException as e:
-                profile = e.profile
-
-                # This isn't strictly necessary, but we might as well use the renewed access token.
-                profile.github_access_token = access_token
-                assert profile.load_fields_from_github(always=True)
-                logging.debug("Swapped to existing profile for existing GitHub user")
+    def post(self, request: Request):
+        """
+        Login if the 'code' parameter is provided. Log out otherwise.
+        """
 
-            profile.save()
-            request.session["profile"] = profile.id
-        elif error == "bad_verification_code":
-            return Response({
-                "error": "Invalid or expired GitHub OAuth verification code",
-            }, status=status.HTTP_401_UNAUTHORIZED)
+        if "code" in request.data:
+            GitHubUser.login(request, request.data["code"])
         else:
-            raise Exception(f"Unknown GitHub login error: {error} - {response['error_description']}")
+            logout(request)
 
-    else:
-        profile.load_fields_from_github()
-        profile.save()
+            profile = Profile()
+            profile.save()
+            request.profile = profile
```

Comment:
I think you also need to store the profile id in the `request.session`? I think as-is, the next request will *also* create a new session?

(Could you return an empty response instead? Or a "logout successful" kind of message?)

Reply:
The idea behind returning the new, logged-out (or logged-in) user is to make the API more RESTful (i.e., you POST to /user to change your current user, and it returns the updated state). I've changed it to return a `message` string with "Login success" or "Logout success" alongside the new `user`.

Reply:
Looks good!